### ISSUE TITLE
Design draft for 2D abstractions

### ISSUE THREAD

**Issue by robinsedlaczek** [dwmkerr/sharpgl#87]
**Summary**
In order to do 2D graphics (e.g. for creating UI controls) there should be an abstraction for 2D graphics in SharpGL. This abstraction should hide the complexity of OpenGL's 3D staff away (e.g. projection, third dimension, lighting etc.) and allow simple 2D drawing operations (e.g. Point, Line, Circle, Rectangle etc.).

**Acceptance Criteria**
1. There is a design draft that describes how the 2D abstraction looks like.
2. There is a first  prototype for the 2D abstraction layer.

**Comment by dwmkerr** [comment]
Agreed, I'd like to see what such a layer would look like. This will be a particularly interesting part of the project.
